react-events-and-state/src/index.jsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class CustomButton extends React.Component{
+    constructor(props){
+        super(props);
+        this.state = {isClicked: false};
+        this.handleClick = this.handleClick.bind(this);
+
+    }
+
+    handleClick(){
+        this.setState(state => ({isClicked: true}));
```

Comment:
While you can pass `setState` a callback function, most of the time it is better to just pass it an object with the properties you want to update. For example:

```jsx
this.setState({ isClicked: true });
```